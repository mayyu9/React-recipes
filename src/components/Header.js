import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import Logo from '../static/images/Logo.jpg';

const HeaderLink = ({ children, ...props }) => (
  <NavLink
    exact
    className="p1 mx2 black rounded text-decoration-none"
    activeClassName="bg-white"
    {...props}
  >
    {children}
  </NavLink>
);


const Header = () => (
  <header className="flex align-center justify-between px4">
    <div>
      <h1 className="h1 inline-block mr2">My Recipes App </h1>
      <span className="mr1">by</span>
      <img src={Logo} alt="logo" style={{ width: 30 }} />
    </div>
    <nav>
      <HeaderLink to="/">Home</HeaderLink>
      <HeaderLink to="/favourites">Favourites</HeaderLink>
    </nav>
  </header>
);

HeaderLink.propTypes = {
  children: PropTypes.node,
};

export default Header;
