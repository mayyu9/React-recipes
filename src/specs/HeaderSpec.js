import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Header from '../components/Header';

describe('<Header />', () => {
  test('should render header component', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
