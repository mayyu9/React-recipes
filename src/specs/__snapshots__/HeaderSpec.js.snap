// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render header component 1`] = `
<header
  className="flex align-center justify-between px4"
>
  <div>
    <h1
      className="h1 inline-block mr2"
    >
      My Recipes App 
    </h1>
    <span
      className="mr1"
    >
      by
    </span>
    <img
      alt="logo"
      src="test-file-stubs"
      style={
        Object {
          "width": 30,
        }
      }
    />
  </div>
  <nav>
    <a
      aria-current="page"
      className="p1 mx2 black rounded text-decoration-none bg-white"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Home
    </a>
    <a
      aria-current={null}
      className="p1 mx2 black rounded text-decoration-none"
      href="/favourites"
      onClick={[Function]}
    >
      Favourites
    </a>
  </nav>
</header>
`;
